src/components/AppSettingsModal/AppSettingsModal.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import produce from 'immer';
+
+import * as actions from '../../actions';
+import { getDefaultProjectPath } from '../../reducers/app-settings.reducer';
+import { COLORS } from '../../constants';
+import { setNested } from '../../utils';
+
+import Modal from '../Modal';
+import ModalHeader from '../ModalHeader';
+import Spacer from '../Spacer';
+import { FillButton } from '../Button';
+import FormField from '../FormField';
+import Toggle from '../Toggle';
+
+import DirectoryPicker from '../DirectoryPicker';
+import ProjectTypeSelection from '../ProjectTypeSelection';
+
+import type { AppSettings } from '../../types';
+
+type Props = {
+  isVisible: boolean,
+  settings: AppSettings,
+  hideModal: () => void,
+  saveAppSettings: (newSettings: AppSettings) => void,
+};
+
+type State = {
+  newSettings: AppSettings,
+};
+
+class AppSettingsModal extends PureComponent<Props, State> {
+  state = {
+    newSettings: this.props.settings,
+  };
+
+  saveSettings = (ev: SyntheticEvent<*>) => {
+    ev.preventDefault();
+
+    const { saveAppSettings } = this.props;
+    const { newSettings } = this.state;
+
+    saveAppSettings(newSettings);
+  };
+
+  // Update setting by key
+  // Notice: Key will be nested e.g. 'general.defaultProjectPath' that's why we're using utility function setNested
+  updateSetting = (key: string, value: any) => {
+    this.setState(state =>
+      // produce could be used with currying here - so no state passed to produce. But Flow couldn't detect the right return type.
+      produce(state, draftState => {
+        setNested(draftState.newSettings, key, value);
+      })
+    );
+  };
+
+  render() {
+    const { hideModal, isVisible } = this.props;
+    const newSettings = this.state.newSettings;
+    return (
+      <Modal isVisible={isVisible} onDismiss={hideModal}>
+        <ModalHeader title="Preferences" />
+
+        <MainContent>
+          <form onSubmit={this.saveSettings}>
+            {/* // todo: Refactor to map over settings so this renders dynamically -
+                         needs additional info on state e.g. component for rendering + props
+                         -> For now it's OK to hard code as we're only having three
+                            settings 
+            */}
+            <SectionTitle>General</SectionTitle>

```

Comment:
@melanieseltzer as mentioned in the todo comment above I think we could create the complete form automatically based on a settings object. We just need to add an info about the component to render & something about the action for saving the settings.
But I think this will make more sense later when we're having more settings. For now it's OK to leave it as it is. I'm imaging something like in the pseudo code below (`capitalize` helper needs to be created):
```js
{ Object.keys(appSettingsConfig).map((sectionName) => (
     <Fragment>
        <SectionTitle>{capitalize(sectionName)}</SectionTitle>
        { Object.keys(appSettingsConfig[SectionName]).map(({saveAction, settingsType, props})=> (
           <SettingComponents[settingType] {...props} onSettingsSaveAction={saveAction}/>    
         ))}
       <Spacer size={10} />
    </Fragment>
  ))
}
```
- `SettingComponents` is a mapping object to map the type to a component.
- `saveAction` is not working here as it's a string in the configuration - needs to call `this.updateSetting`.

- `appSettingsConfig` will be a configuration file with the same object structure as the `app-settings` state but containing info about the rendering (we could add this to state as well but feels like it would mess/clutter the state).
**`appSettingsConfig` example:**
```js
export const appSettings = {
  general: {
    defaultProjectPath: {
      settingsType: 'pathPicker',
      props: '...',
      saveAction: '...'
    },
    defaultProjectType: {
      setingsType: 'projectType',
      ...
    },
  },
  privacy: {
    enableUsageTracking: {
      settingsType: 'toggle',
      ...
    },
  },
};
```

Reply:
Yeah this sounds good. Do you want to create a new issue with this info for later? Don't want it to get lost in the PR review.

Reply:
Yes, a new issue would be great. Can you please copy it into a new issue? And maybe you can add something how the `saveAction` could be handled. I haven't thought about that yet.